File: 20190828-test/20190828-test/ViewController.m
//
//  ViewController.m
//  20190828-test
//
//  Created by dongouc on 2019/8/28.
//  Copyright © 2019 dongouc. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    dispatch_async(dispatch_get_global_queue(0, 0), ^{
        NSLog(@"%s， %d", __func__, __LINE__);
        dispatch_async(dispatch_get_main_queue(), ^{
            NSLog(@"%s， %d", __func__, __LINE__);
        });
        NSLog(@"%s， %d", __func__, __LINE__);
    });
    NSLog(@"%s， %d", __func__, __LINE__);
}


@end

File: selfTestObj/selfTestObj/ViewController.swift
//
//  ViewController.swift
//  selfTestObj
//
//  Created by dongouc on 2019/8/4.
//  Copyright © 2019 dongouc. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        test001()
    }
    

}

extension ViewController {
    
    /// 单链表翻转实验
    func test001() {
        let testListNode = ListNode(1, ListNode(2, ListNode(3, nil)))
        
        listNodeValuePrint(testListNode)
        debugPrint("============")
        listNodeValuePrint(reverseList01(testListNode)!)
        debugPrint("============")
        
        let testListNode2 = ListNode(1, ListNode(2, ListNode(3, nil)))
        listNodeValuePrint(testListNode2)
        debugPrint("============")
        listNodeValuePrint(reverseList02(testListNode2)!)
    }
}
